client/question/services/FeedbackGeneratorService.js

```diff
@@ -62,6 +95,38 @@ tie.factory('FeedbackGeneratorService', [
     var previousRuntimeErrorString = '';
 
     /**
+     * Pseudo-randomly returns an int between min (inclusive) and max.
+     *
+     * @param {number} min Minimum number for range (inclusive).
+     * @param {number} max Maximum number for range (exclusive).
+     * @returns {number}
+     * @private
+     */
+    var _getRandomInt = function(min, max) {
+      var minimum = Math.ceil(min);
+      var maximum = Math.floor(max);
+      return Math.floor(Math.random() * (maximum - minimum)) + minimum;
+    };
+
+    /**
+     * Randomly selects feedback string from an object containing variations
+     * of a given feedback type.
+     *
+     * @param {string} type Correctness feedback type.
+     * @returns {string}
+     * @private
+     */
+    var _getCorrectnessFeedbackString = function(type) {
+      var correctnessFeedbackIndex = previousCorrectnessFeedback[type];
+      while (correctnessFeedbackIndex === previousCorrectnessFeedback[type]) {
```

Comment:
This is ... OK, I guess. It would be slightly better to generate a random int from [0, arr.length - 1] and adjust the index accordingly.